File: 1544-make-the-string-great/1544-make-the-string-great.java
class Solution {
    public String makeGood(String s) {
  while (true) {
         boolean flag = false;
         for (int i = 0; i < s.length() - 1; i++) {
             if (Math.abs(s.charAt(i) - s.charAt(i + 1)) == 32) {
                 s = s.substring(0, i) + s.substring(i + 2);
                 flag = true;
                 break;
             }
         }
         if (!flag) break;
     }
        return s;
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        n=len(s)
        stk=[]
        for i in range(len(s)):
            if stk and abs(ord(stk[-1])-ord(s[i]))==32:
                stk.pop()
            else:
                stk.append(s[i])
        return "".join(stk)
                
            